src/js/engagement_view/src/setupProxy.js

```diff
@@ -0,0 +1,18 @@
+const { createProxyMiddleware } = require('http-proxy-middleware');
```

Comment:
Is the change from `*.tsx` to `*.js` in the filename significant? This seems (to my admittedly shallow knowledge of the dark inner workings of Javascript and Typescript) to be the same file but with a single additional proxy middleware.

Reply:
Unfortunately, it is. It must be. I spent **far** too much time thinking it should work as a Typescript file, but no dice.

Reply:
That's probably worth an explanatory comment to that effect, as well as any link to relevant background information or documentation.